dvc/fs/azure.py

```diff
@@ -116,8 +116,21 @@ def _prepare_credentials(self, **config):
         login_info["client_id"] = config.get("client_id")
         login_info["client_secret"] = config.get("client_secret")
 
-        if not any(
+        if not (login_info["account_name"] or login_info["connection_string"]):
+            raise AzureAuthError(
+                "Authentication to Azure Blob Storage requires either "
+                "account_name or connection_string.\nLearn more about "
+                "configuration settings at "
+                + format_link("https://man.dvc.org/remote/modify")
```

Comment:
I'm not 100% sure I get this change (see Q in https://github.com/iterative/dvc.org/pull/2489#pullrequestreview-665102806) but maybe:
```suggestion
                "Authentication to Azure Blob Storage requires setting either "
                "account_name or connection_string.\nLearn more about "
                "remote storage settings at "
                + format_link("https://man.dvc.org/remote/modify")
```

Some quick explanation of the change in the PR desc. would be great. Or link to a specific comment in #5797